### ISSUE TITLE
Move to py.test

### ISSUE THREAD

**Issue by GaelVaroquaux** [scipy-lectures/scipy-lecture-notes#296]
Nosetests is no longer maintained, and py.test has a nicer runner. We should consider moving to py.tests

**Comment by pdebuyl** [comment]
Fow now, I have good success with running

    python3 -m pytest --maxfail 2 --doctest-glob '*.rst' --ignore advanced

that runs the doctests.

I ignore `advanced` for now because in it compiled extensions need to be build. It is only a matter of time prioritization, I couldn't get there yet :-)

**Comment by drvinceknight** [comment]
They seem to hang for me with a matplotlib figure being open (ubuntu). If I can figure out which figure that is, it's easy to fix: just need to close it.

**Comment by pdebuyl** [comment]
python 2 or 3?

**Comment by pdebuyl** [comment]
nevermind, they both do it actually

**Comment by drvinceknight** [comment]
```
py.test $(shell find intro advanced packages -name \*.rst -print) 
```

this no longer hangs but various failures appear. Hunting them down now :) Probably just need to specify the options right. :+1:

**Comment by pdebuyl** [comment]
Almost there in #321 :-) Only compiled extensions are omitted.

**Comment by GaelVaroquaux** [comment]
Fixed in #321  and #327. Thanks @pdebuyl !
